internal/service/ipam/api_address_block_resource_test.go

```diff
@@ -0,0 +1,1126 @@
+	})
+}
+
+func TestAccAddressBlockResource_HostnameRewriteChar(t *testing.T) {
+	var resourceName = "bloxone_ipam_address_block.test_hostname_rewrite_char"
+	var v ipam.IpamsvcAddressBlock
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAddressBlockHostnameRewriteChar("192.168.0.0", "16", "a"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "hostname_rewrite_char", "a"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccAddressBlockHostnameRewriteChar("192.168.0.0", "16", "c"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "hostname_rewrite_char", "c"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccAddressBlockResource_HostnameRewriteEnabled(t *testing.T) {
+	var resourceName = "bloxone_ipam_address_block.test_hostname_rewrite_enabled"
+	var v ipam.IpamsvcAddressBlock
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAddressBlockHostnameRewriteEnabled("192.168.0.0", "16", "true"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "hostname_rewrite_enabled", "true"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccAddressBlockHostnameRewriteEnabled("192.168.0.0", "16", "false"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "hostname_rewrite_enabled", "false"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccAddressBlockResource_HostnameRewriteRegex(t *testing.T) {
+	var resourceName = "bloxone_ipam_address_block.test_hostname_rewrite_regex"
+	var v ipam.IpamsvcAddressBlock
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAddressBlockHostnameRewriteRegex("192.168.0.0", "16", "[^a-z]"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "hostname_rewrite_regex", "[^a-z]"),
+				),
+			},
+			// Update and read
+			// Create and Read
```

Comment:
extra line copied by mistake?

Reply:
Yes

Reply:
I think you should have kept the `Update and read` and deleted the `Create and read`